最新仓库的 b5be05 自构建 docker image 失败： The `npm ci` command can only install with an existing package-lock.json

### 路由地址

/rsshub/routes

### 完整路由地址

/rsshub/routes

### 相关文档

https://docs.rsshub.app/install/

### 预期是什么？

尝试使用 `docker compose up -d --build ` 编译镜像并运行。

### 实际发生了什么？

```bash
➜  rsshub docker-compose up -d --build
Sending build context to Docker daemon  1.306MB
Step 1/24 : FROM node:14-buster-slim as dep-builder
 ---> eb79fdb26bfb
Step 2/24 : LABEL MAINTAINER https://github.com/DIYgod/RSSHub/
 ---> Using cache
 ---> c4749b04b7de
Step 3/24 : ARG USE_CHINA_NPM_REGISTRY=0;
 ---> Using cache
 ---> c1c517d40271
Step 4/24 : ARG PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=1;
 ---> Using cache
 ---> 444118db784f
Step 5/24 : RUN ln -sf /bin/bash /bin/sh
 ---> Using cache
 ---> 14503b0f1f74
Step 6/24 : RUN apt-get update && apt-get install -yq libgconf-2-4 apt-transport-https git dumb-init python3 build-essential --no-install-recommends
 ---> Using cache
 ---> 2fdbbbe8f179
Step 7/24 : WORKDIR /app
 ---> Using cache
 ---> 54adccefa424
Step 8/24 : COPY . /app
 ---> 16bf21ba9903
Step 9/24 : RUN if [ "$USE_CHINA_NPM_REGISTRY" = 1 ]; then   echo 'use npm mirror'; npm config set registry https://registry.npm.taobao.org;   fi;
 ---> Running in 50c555765df0
 ---> 1bca805ecd7c
Step 10/24 : RUN npm i -g npm
 ---> Running in 01837ca3d36d
/usr/local/bin/npm -> /usr/local/lib/node_modules/npm/bin/npm-cli.js
/usr/local/bin/npx -> /usr/local/lib/node_modules/npm/bin/npx-cli.js
+ npm@8.3.0
added 67 packages from 17 contributors, removed 290 packages and updated 147 packages in 11.3s
 ---> 9d89a70cda08
Step 11/24 : RUN if [ "$PUPPETEER_SKIP_CHROMIUM_DOWNLOAD" = 0 ]; then   unset PUPPETEER_SKIP_CHROMIUM_DOWNLOAD && npm ci ;  else   export PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true && npm ci ;  fi;
 ---> Running in 25d48fc7f353
npm ERR! The `npm ci` command can only install with an existing package-lock.json or
npm ERR! npm-shrinkwrap.json with lockfileVersion >= 1. Run an install with npm@5 or
npm ERR! later to generate a package-lock.json file, then try again.

npm ERR! A complete log of this run can be found in:
npm ERR!     /root/.npm/_logs/2021-12-12T07_51_56_694Z-debug-0.log
1 error occurred:
        * Status: The command '/bin/sh -c if [ "$PUPPETEER_SKIP_CHROMIUM_DOWNLOAD" = 0 ]; then   unset PUPPETEER_SKIP_CHROMIUM_DOWNLOAD && npm ci ;  else   export PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true && npm ci ;  fi;' returned a non-zero code: 1, Code: 1
```

似乎是在 Dockerfile 的这一步发生了错误

```
RUN if [ "$PUPPETEER_SKIP_CHROMIUM_DOWNLOAD" = 0 ]; then \
  unset PUPPETEER_SKIP_CHROMIUM_DOWNLOAD && npm ci ;\
  else \
  export PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true && npm ci ;\
  fi;
```


### 部署

自建

### 部署相关信息

Docker version 20.10.11, build dea9396 Docker Compose version v2.1.1

### 额外信息

```shell
docker-compose.yml

version: '3'
services:
  rsshub:
    build: ./RSSHub
    container_name: rsshub
    restart: unless-stopped
    ports:
      - 1200:1200
    environment:
      - NODE_ENV=production
      - CACHE_TYPE=null
      - PUPPETEER_WS_ENDPOINT=ws://browserless:3000
      - DUBUG_INFO=false
      - TIME_ZONE=Asia/Shanghai
    env_file:
      - ./rsshub.env
    depends_on:
      - browserless

  browserless:
    image: browserless/chrome
    restart: unless-stopped
    ports:
      - "30001:3000"

networks:
  default:
    external: true
    name: net


```

```
➜  rsshub tree -L 1 ./
./
|-- RSSHub
|-- docker-compose.yml
`-- rsshub.env

1 directory, 2 files
```

### 这不是重复的 issue

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues)，以确保该错误尚未被报告。

--- a/Dockerfile
+++ b/Dockerfile
@@ -10,7 +10,11 @@ RUN ln -sf /bin/bash /bin/sh
 RUN apt-get update && apt-get install -yq libgconf-2-4 apt-transport-https git dumb-init python3 build-essential --no-install-recommends
 
 WORKDIR /app
-COPY . /app
+
+COPY ./yarn.lock /app
+COPY ./package.json /app
+
+#COPY . /app
 
 RUN if [ "$USE_CHINA_NPM_REGISTRY" = 1 ]; then \
   echo 'use npm mirror'; npm config set registry https://registry.npm.taobao.org; \
@@ -19,10 +23,12 @@ RUN if [ "$USE_CHINA_NPM_REGISTRY" = 1 ]; then \
 RUN npm i -g npm
 
 RUN if [ "$PUPPETEER_SKIP_CHROMIUM_DOWNLOAD" = 0 ]; then \
-  unset PUPPETEER_SKIP_CHROMIUM_DOWNLOAD && npm ci ;\
+  unset PUPPETEER_SKIP_CHROMIUM_DOWNLOAD && yarn ;\
   else \
-  export PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true && npm ci ;\
+  export PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true && yarn ;\
   fi;
+
+COPY . /app
 
 RUN node scripts/docker/minify-docker.js
 

--- a/Dockerfile
+++ b/Dockerfile
@@ -14,7 +14,6 @@ WORKDIR /app
 COPY ./yarn.lock /app
 COPY ./package.json /app
 
-#COPY . /app
 
 RUN if [ "$USE_CHINA_NPM_REGISTRY" = 1 ]; then \
   echo 'use npm mirror'; npm config set registry https://registry.npm.taobao.org; \

